src/core/algorithms/nd/nd.h

```diff
@@ -25,6 +27,14 @@ class ND {
         return rhs_;
     }
 
+    [[nodiscard]] std::vector<std::string> GetLhsNames() const {
+        return GetVerticalNames(GetLhs());
+    }
+
+    [[nodiscard]] std::vector<std::string> GetRhsNames() const {
+        return GetVerticalNames(GetRhs());
+    }
```

Comment:
After moving the `GetVerticalNames()` implementation of these methods should be placed in .cpp file.